presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusPredicateTimeInfo.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.prometheus;
+
+import java.time.ZonedDateTime;
+import java.util.Optional;
+
+public class PrometheusPredicateTimeInfo
+{
+    Optional<ZonedDateTime> predicateLowerTimeBound;
+    Optional<ZonedDateTime> predicateUpperTimeBound;
+
+    private PrometheusPredicateTimeInfo(Builder builder)
+    {
+        predicateLowerTimeBound = builder.predicateLowerTimeBound;
+        predicateUpperTimeBound = builder.predicateUpperTimeBound;
+    }
+
+    public Optional<ZonedDateTime> getPredicateLowerTimeBound()
+    {
+        return predicateLowerTimeBound;
+    }
+
+    public Optional<ZonedDateTime> getPredicateUpperTimeBound()
+    {
+        return predicateUpperTimeBound;
+    }
+
+    public Builder toBuilder()
+    {
+        return new Builder(this);
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static final class Builder
```

Comment:
I think this builder is not needed. It does not have any setter.

Reply:
To my mind the builder pattern here: encapsulates the usage of PrometheusPredicateTimeInf; allows it to built in steps in `PrometheusSplitManager` and it's immutable, so no setters.